src/foundry/foundry.js/clientDocuments/chatMessage.d.ts

```diff
@@ -1,16 +1,221 @@
+        | InstanceType<CONFIG['Token']['objectClass']>;
+      alias?: string;
+    }): ChatSpeakerData;
+
+    /**
+     * A helper to prepare the speaker object based on a target TokenDocument
+     *
+     * @param token - The TokenDocument of the speaker
+     * @param alias - The name of the speaker to display
+     * @returns The identified speaker data
+     */
+    protected static _getSpeakerFromToken({
+      token,
+      alias
+    }: {
+      token: InstanceType<ConfiguredDocumentClass<typeof BaseToken>> | InstanceType<CONFIG['Token']['objectClass']>;
+      alias: string;
+    }): ChatSpeakerData;
+
+    /**
+     * A helper to prepare the speaker object based on a target Actor
+     *
+     * @param scene - The Scene is which the speaker resides
+     * @param actor - The Actor that is speaking
+     * @param alias - The name of the speaker to display
+     * @returns The identified speaker data
+     */
+    protected static _getSpeakerFromActor({
+      scene,
+      actor,
+      alias
+    }: {
+      scene?: InstanceType<ConfiguredDocumentClass<typeof Scene>>;
+      actor?: InstanceType<ConfiguredDocumentClass<typeof Actor>>;
+      alias?: string;
+    }): ChatSpeakerData;
+
+    /**
+     * A helper to prepare the speaker object based on a target User
+     *
+     * @param scene - The Scene in which the speaker resides
+     * @param user  - The User who is speaking
+     * @param alias - The name of the speaker to display
+     * @returns The identified speaker data
+     */
+    protected static _getSpeakerFromUser({
+      scene,
+      user,
+      alias
+    }: {
+      scene?: InstanceType<ConfiguredDocumentClass<typeof Scene>>;
+      user?: InstanceType<ConfiguredDocumentClass<typeof User>>;
+      alias?: string;
+    }): ChatSpeakerData;
+
+    /**
+     * Obtain an Actor instance which represents the speaker of this message (if any)
+     * @param speaker - The speaker data object
+     */
+    static getSpeakerActor(speaker: ChatSpeakerData): Actor | null;
+
+    /* -------------------------------------------- */
+
+    /**
+     * Obtain a data object used to evaluate any dice rolls associated with this particular chat message
+     */
+    getRollData(): InstanceType<ConfiguredDocumentClass<typeof Actor>>['getRollData'];
+
+    /**
+     * Given a string whisper target, return an Array of the user IDs which should be targeted for the whisper
+     *
+     * @param name - The target name of the whisper target
+     * @returns An array of User instances
+     */
+    static getWhisperRecipients(name: string): InstanceType<ConfiguredDocumentClass<typeof User>>[];
+
+    /**
+     * Render the HTML for the ChatMessage which should be added to the log
+     */
+    getHTML(): Promise<JQuery>;
+
+    /** @override */
+    _preCreate(
+      data: Parameters<ChatMessageData['_initializeSource']>[0],
+      options: DocumentModificationOptions,
+      user: BaseUser
```

Comment:
I think whenever we are inside of `foundry.js`, it's better to refer to the base classes via `foundry.documents.` instead of importing them because that's also what foundry does to refer to them.